server/channels/store/sqlstore/user_store.go

```diff
@@ -2268,9 +2268,50 @@ func (us SqlUserStore) RefreshPostStatsForUsers() error {
 	return nil
 }
 
+func applyUserReportFilter(query sq.SelectBuilder, filter *model.UserReportOptions, isPostgres bool) sq.SelectBuilder {
+	query = applyRoleFilter(query, filter.Role, isPostgres)
+	if filter.HasNoTeam {
+		query = query.Where(sq.Expr("u.Id NOT IN (SELECT UserId FROM TeamMembers WHERE DeleteAt = 0)"))
+	} else if filter.Team != "" {
+		query = query.Join("TeamMembers tm ON (tm.UserId = u.Id AND tm.DeleteAt = 0)").
+			Where(sq.Eq{"tm.TeamId": filter.Team})
+	}
+	if filter.HideActive {
+		query = query.Where(sq.Gt{"u.DeleteAt": 0})
+	}
+	if filter.HideInactive {
+		query = query.Where(sq.Eq{"u.DeleteAt": 0})
+	}
+
+	if strings.TrimSpace(filter.SearchTerm) != "" {
```

Comment:
Can we trim the input in API layer itself so that we get trimmed input everywhere?

Reply:
This is what's being done in the store for other searching methods, so I'd like to replicate it the same here to keep consistency, since we expect other methods to do the same.

Reply:
Okay.